publication-model/src/main/java/no/unit/nva/model/contexttypes/Book.java

```diff
@@ -143,6 +147,19 @@ private List<String> extractValidIsbnList(List<String> isbnList) {
                    .toList();
     }
 
+    @JsonIgnore
+    @Override
+    public Set<URI> extractPublicationContextUris() {
+        var uris = new HashSet<URI>();
+        if (nonNull(publisher) && publisher instanceof Publisher publisherWithId && nonNull(publisherWithId.getId())) {
+            uris.add(publisherWithId.getId());
+        }
+        if (nonNull(series) && series instanceof Series seriesWithId && nonNull(seriesWithId.getId())) {
+            uris.add(seriesWithId.getId());
+        }
+        return Collections.unmodifiableSet(uris);
+    }
```

Comment:
Now it looks like it will make sense to make `Publisher`, `Series` and `Journal` to implement class `PublicationChannel` with `getId()` method.